### PR TITLE
WIP Add new PTB example

### PR DESCRIPTION
**Pull request by baggepinnen**
I am working on a new example. The original python tutorial can be found here
https://www.tensorflow.org/versions/r0.10/tutorials/recurrent/index.html

The code is currently not working, I thought it could be used to locate bugs. Right now it fails a lot around lines 150-167 in `ptb_word_lm.jl`

### CODE REVIEW

**Comment by malmaud** [comment]
Thanks for this! This is a particularly good test since the RNN code has largely been reimplemented in Julia (as it is defined in the Python layer in the Python TensorFlow) so bugs especially likely.

File: examples/ptb_word_lm.jl [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+    end
+
+    inputs = TensorFlow.split(1, num_steps, inputs);
+    outputs, state = nn.rnn(cell, inputs, initial_state=initial_state);
+
+    output = reshape(concat(1, outputs), [-1, size])
+    softmax_w = get_variable("softmax_w", [size, vocab_size], Float32)
+    softmax_b = get_variable("softmax_b", [vocab_size], Float32)
+    logits = matmul(output, softmax_w) + softmax_b
+    loss = nn.seq2seq.sequence_loss_by_example([logits],[reshape(targets, [-1])],[ones([batch_size * num_steps], Float32)])
```
Comment (malmaud) [inline examples/ptb_word_lm.jl]: This is not a function we have right now. Might be useful to implement though.

File: examples/ptb_word_lm.jl [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+    final_state = state
+
+    if !is_training
+        return
+    end
+
+
+    lr = Variable(0.0, trainable=false)
+    tvars = trainable_variables()
+    grads, _ = clip_by_global_norm(gradients(cost,tvars), config.max_grad_norm)
```
Comment (malmaud) [inline examples/ptb_word_lm.jl]: Another function I haven't implemented yet.

**Comment by kshyatt** [comment]
Is this working now?

**Comment by baggepinnen** [comment]
The PTB example as a whole is not working yet. I will resume work on it within a few weeks, other stuff had to take priority for a while :/
I'm happy to see that some of the missing pieces are now in place, which will surely speed up the development here.

**Comment by malmaud** [comment]
In particular this should wait for https://github.com/malmaud/TensorFlow.jl/pull/81.

**Comment by malmaud** [comment]
As well as https://github.com/malmaud/TensorFlow.jl/issues/82

**Comment by malmaud** [comment]
All the machinery should be in place for this now, if you want to check if things are working.

**Comment by baggepinnen** [comment]
I'll have a look at this soon again, awaiting some experience from a colleague working on a similar implementation.

**Comment by malmaud** [comment]
A lot of the RNN API has changed since this was first submitted, so I'm going as outdated for the moment.

**Pull request closed without merging**